Added filter options for Difficulty and Category in algorithm list

ADITHICJ opened the pull request:
## Description
This PR implements filter options for Difficulty and Category in the Algorithm Explorer, helping users easily discover algorithms based on their learning preferences and skill levels.

Previously, users could only perform a basic text search, making it difficult to focus on topics that match their experience level or area of interest.

Fixes #201

## Type of Change
- [ ] Bug fix
- [x] New feature  
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Other

## Testing
How has this been tested?
- [x] Manual testing
- [ ] Added/updated tests
- [x] Tested on multiple browsers/devices

## Screenshots (if applicable)
<img width="1180" height="247" alt="image" src="https://github.com/user-attachments/assets/23f05819-8f1d-4bf7-9c10-d7cab3bd1176" />
<img width="1468" height="678" alt="image" src="https://github.com/user-attachments/assets/5169be02-21a0-4ae0-a30d-53b87c5d19b8" />
<img width="1508" height="685" alt="image" src="https://github.com/user-attachments/assets/6337235e-4e58-43bb-8f24-70ade1ed4ae0" />


## Checklist
- [x] Code follows project style guidelines
- [x] Self-reviewed my code  
- [ ] Updated documentation if needed
- [x] No new warnings or errors
- [x] Tested changes locally

## Hacktoberfest (if applicable)
- [x] This is a meaningful contribution (not spam/low-effort)
- [x] I have read the contribution guidelines

Resolves #201

Harshdev625 commented:
@ADITHICJ It’s not working properly in my Codespace. Please contact me via DM on Discord for further details: [https://discord.com/channels/707751027973161132/1426969903034007643](https://discord.com/channels/707751027973161132/1426969903034007643)

ADITHICJ commented:
@Harshdev625 While implementing the filtering feature, I also made minor updates to the Algorithm model schema.

- Added the category field (array of strings)
- Added the difficulty field (enum: "Easy", "Medium", "Hard")

These fields are now required for the filtering functionality to work correctly on both frontend and backend.

If existing algorithm entries are missing these fields, they may need to be updated or re-seeded with the new structure.
